streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java

```diff
@@ -85,8 +85,23 @@
      * @throws IllegalStateException If store gets registered after initialized is already finished
      * @throws StreamsException if the store's change log does not contain the partition
      */
+    default void register(final StateStore store,
+                  final StateRestoreCallback stateRestoreCallback) {
+        register(store, stateRestoreCallback, null);
+    }
+
+    /**
+     * Register and possibly restores the specified storage engine.
+     *
+     * @param store the storage engine
+     * @param stateRestoreCallback the restoration callback logic for log-backed state stores upon restart
+     * @param checkpointCallback called to checkpoint position metatadata of state stores
+     *
+     * @throws IllegalStateException If store gets registered after initialized is already finished
+     * @throws StreamsException if the store's change log does not contain the partition
+     */
     void register(final StateStore store,
-                  final StateRestoreCallback stateRestoreCallback);
+                  final StateRestoreCallback stateRestoreCallback, final CheckpointCallback checkpointCallback);
```

Comment:
This is a public API change. We need a KIP for this. \cc @vvcephei

Reply:
Absolutely, thanks for the reminder @mjsax !